### ISSUE TITLE
Cannot use `DataFrames.getindex` on column of type `NullableList{String,Int32,Primitive{UInt8}}`

### ISSUE THREAD

**Issue by metanoid** [ExpandingMan/Arrow.jl#47]
I have a `DataFrame` (`df`) which was constructed from a `.feather` file using `Feather.jl`.  The data is strings, with some missing values.

One of the columns (column `9`) is of type `NullableList{String,Int32,Primitive{UInt8}}`.

When I call `df[:, 9]` I get the following error message:

```julia
julia> df[:,j]
ERROR: UndefVarError: p not defined
Stacktrace:
 [1] NullableList{String,K,P} where P<:AbstractPrimitive where K<:Integer(::NullableList{String,Int32,Primitive{UInt8}}) at C:\Users\username\.julia\packages\Arrow\b4oSO\src\lists.jl:258
 [2] Type at C:\Users\username\.julia\packages\Arrow\b4oSO\src\lists.jl:260 [inlined]
 [3] copy(::NullableList{String,Int32,Primitive{UInt8}}) at C:\Users\username\.julia\packages\Arrow\b4oSO\src\lists.jl:262
 [4] getindex(::DataFrame, ::Colon, ::Int64) at C:\Users\username\.julia\packages\DataFrames\0Em9Q\src\dataframe\dataframe.jl:359
 [5] top-level scope at none:0
```

To me, it looks as if the problem is in `lists.jl`, lines 258 to 260, where the argument to the function is called `l` but is referred to in the body of the function as `p`.  But I don't know for sure.

**Comment by ExpandingMan** [comment]
Thanks, it's quite unfortunate that this passed unit tests.  For now I've just made a very quick fix, see #48.  If you can verify that this fixes your case, it would be appreciated.  Otherwise, I will merge the PR soon and issue a patch release tag.

**Comment by metanoid** [comment]
I confirm that #48 fixes the issue on my end, thanks
